### DISCUSSION TITLE
FYI: Big Wolfram 14 upgrade (plus many other upgrades) required a much bigger extend

### DISCUSSION THREAD

**Discussion by tinker2much** [gitbls/sdm#205] [General]
(This isn't an error report, just an interesting case, I think...)

In my typical sdm invocations, I have sdm install a bunch of software, and also do a dist-upgrade, so I always extend the images to create enough space. On my full images, I normally do the following (extending the image by 2GB): 
`--extend --xmb 2048` 

When starting with the stock 2024-03-12 images, a _really_ big Wolfram upgrade gets pulled in now during the upgrade, and I found that I had to bump up the extend by _another_ 3GB, for a total of 5GB:
`--extend --xmb 5120`

I roughly binary-searched my way between extend _totals_ of 2GB and 8GB, and I found that while 4GB wasn't big enough, 5GB was. I didn't try to fix the answer any finer than that, so the real required _total_ is somewhere between 4GB and 5GB, and the required _additional_ amount is something between 2 and 3 _additional_ GB. 

I looked at /etc/sdm/apt.log and /etc/sdm/history in the image, finding that a total of 79 packages needed upgrading, and that the dist-upgrade used up 1.9GB!:
```
2024-04-30 17:40:15 > IMG '/home/pi/Downloads/bookworm64-full.img' has 5885872 1K-blocks (6.0GB, 5.6GiB) free at start of 'apt dist-upgrade'
2024-04-30 17:44:57 > IMG '/home/pi/Downloads/bookworm64-full.img' has 3994084 1K-blocks (4.1GB, 3.8GiB) free at end of 'apt dist-upgrade'
```
Just an FYI, like I said...

**Comment by gitbls** [comment]
Thanks for the heads up.

Note that you can also extend it significantly, do the install, and then shrink it afterwards if desired.
